### ISSUE TITLE
False positive issue - MatchingDeclarationName in Jetpack Compose

### ISSUE THREAD

**Issue by Abhimanyu14** [detekt/detekt#4917]
## Expected Behavior
If a have a top-level Composable which matches the name of the file, this issue should not be reported.

## Observed Behavior
If a have a top-level Composable which matches the name of the file, this issue is reported as Composables are not classes.

## Steps to Reproduce

**SettingsScreenView.kt**
```
data class SettingsScreenViewData(
   // arguments
)

@Composable
fun SettingsScreenView() {
}
```

## Context
Compose project give lot of false positives

## Your Environment

* Version of detekt used: 
```
detektVersion = "1.20.0"
detektFormattingVersion = "1.20.0"
detektComposeVersion = "1.0.0"
```

* Version of Gradle used (if applicable): `gradleBuildToolsVersion = "7.2.1"`
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: `Mac OS Monterey 12.4`
* Link to your project (if it's a public repository): https://github.com/Abhimanyu14/finance-manager

**Comment by cortinico** [comment]
The rule is triggered as you effectively have a top level class declaration: 
`data class SettingsScreenViewData` 

So the file should be called `SettingsScreenViewData.kt`.

If you move the data class to a separate file, then you won't be facing this report any more.
